pkg/customers/address.go

```diff
@@ -108,49 +144,71 @@ func updateCustomerAddress(logger log.Logger, repo CustomerRepository) http.Hand
 		}
 
 		if req.Type == "primary" {
-			cust, err := repo.GetCustomer(customerID, organization)
-			if err != nil {
-				moovhttp.Problem(w, err)
-				return
+			var addresses []client.Address
+			if ownerType == client.OWNERTYPE_CUSTOMER {
+				cust, err := repo.GetCustomer(customerID, organization)
+				if err != nil {
+					moovhttp.Problem(w, err)
+					return
+				}
+				addresses = cust.Addresses
+			} else {
+				rep, err := repo.GetCustomerRepresentative(representativeID)
+				if err != nil {
+					moovhttp.Problem(w, err)
+					return
+				}
+				addresses = rep.Addresses
+
 			}
 
-			for _, addr := range cust.Addresses {
+			for _, addr := range addresses {
 				if addr.Type == "primary" && addr.AddressID != addressId {
 					moovhttp.Problem(w, ErrAddressTypeDuplicate)
 					return
 				}
 			}
 		}
 
-		if err := repo.updateCustomerAddress(customerID, addressId, req); err != nil {
-			logger.LogErrorf("error updating customer's address: customer=%s address=%s: %v", customerID, addressId, err)
+		if err := repo.updateAddress(ownerID, addressId, ownerType, req); err != nil {
+			logger.LogErrorf("error updating %s's address: %s=%s address=%s: %v", string(ownerType), string(ownerType), ownerID, addressId, err)
 			moovhttp.Problem(w, err)
 			return
 		}
 
-		logger.Logf("updating address=%s for customer=%s", addressId, customerID)
+		logger.Logf("updating address=%s for %s=%s", addressId, string(ownerType), customerID)
 
 		respondWithCustomer(logger, w, customerID, organization, requestID, repo)
 	}
 }
 
-func deleteCustomerAddress(logger log.Logger, repo CustomerRepository) http.HandlerFunc {
+func deleteAddress(logger log.Logger, ownerType client.OwnerType, repo CustomerRepository) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		w = route.Responder(logger, w, r)
 
 		customerID, addressId := route.GetCustomerID(w, r), getAddressId(w, r)
-		if customerID == "" || addressId == "" {
```

Comment:
I think if we're breaking up this conditional we should break up the line above it too. Something like:
```go
customerID := route.GetCustomerID(w, r)
if customerID == "" {
    return
}

addressID := getAddressId(w, r)
if addressID == "" {
    return
}
```

That existing method `getAddressId()` should probably be renamed to `getAddressID()`.